### PR TITLE
🛢️ feature/bulk annotation

### LINKED ISSUE #4050: Bulk annotation in Feedback datasets
This initial version of the bulk annotation will only apply to the `Pending` queue.

```[tasklist]
### Tasks
- [ ] https://github.com/argilla-io/argilla/pull/4334
- [ ] https://github.com/argilla-io/argilla/pull/4333
- [ ] https://github.com/argilla-io/argilla/issues/4367
- [ ] https://github.com/argilla-io/argilla/issues/4449
```

### LINKED ISSUE #4419: [BUG-UI/UX] paginating to a new record, automatically scrolls down to selected form area
**Describe the bug**
Not sure if this is a bug or a feature related to the usage of the auto-focus on the form area, but maybe we don't want to automatically scroll to the selected area either because it messes a bit with the annotation flow.

**To Reproduce**
1. Go to a dataset
2. Click on open screen halfway
3. Scroll down to go to the next record

**Expected behavior**
Ideally I want to stay at the top of the screen/start of the form because you are not able to annotate anything without reading the fields.

**Screenshots**
https://github.com/argilla-io/argilla/assets/25269220/feb9ea3b-847d-421a-9898-d43f37e9d269

**Environment (please complete the following information):**
 - OS [e.g. iOS]:N.A.
 - Browser [e.g. chrome, safari]: Firefox
 - Argilla Version [e.g. 1.0.0]: 1.20
 - ElasticSearch Version [e.g. 7.10.2]:N.A.
 - Docker Image (optional) [e.g. argilla:v1.0.0]:1.20

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by damianpumar**
- [x] bulk/focus switch
- [x] BulkAnnotation component
- [x] FocusAnnotation component
- [x] Make the header of record fixed
- [x] Include checkbox for each record
- [x] Include global checkbox
- [x] Include page size selector
- [x] Pagination with more than 10 options
- [x] Show selected records number in the form
- [x] Bulk action: Submit / Discard / Save as draft

### CODE REVIEW

File: frontend/components/feedback-task/container/fields/Record.vue [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    border-radius: $border-radius-m;
+  }
+  &__fixed-header {
+    padding: $base-space * 2;
+    & + :deep(.record__content) {
+      padding-top: 0;
+    }
+  }
+}
+// .record {
```
Comment (damianpumar) [inline frontend/components/feedback-task/container/fields/Record.vue]: Could we remove these comments ?

File: frontend/components/feedback-task/container/mode/BulkAnnotation.vue [hunk 2]
```diff
@@ -91,6 +99,7 @@ export default {
   data() {
     return {
       selectedRecords: [],
+      fixedHeight: false,
```
Comment (damianpumar) [inline frontend/components/feedback-task/container/mode/BulkAnnotation.vue:104]: I don't understand this prop :)

File: frontend/pages/dataset/_id/annotation-mode/index.vue [hunk 3]
```diff
@@ -7,8 +7,9 @@
           :key="datasetId"
           :datasetId="datasetId"
           :breadcrumbs="breadcrumbs"
-          :showTrainButton="isAdminOrOwnerRole"
```
Comment (damianpumar) [inline frontend/pages/dataset/_id/annotation-mode/index.vue]: Be careful Leire, this commit is changing the behaviour of the component and some tests are failing.
Why is this component receiving props that it didn't need before?

File: frontend/components/commons/header/header-feedback-task/HeaderFeedbackTask.component.vue [hunk 4]
```diff
@@ -11,7 +11,7 @@
         ref="trainButtonRef"
         class="header__button small"
         @on-click="onClickTrain"
-        v-if="isAdminOrOwnerRole && showTrainButton"
```
Comment (damianpumar) [inline frontend/components/commons/header/header-feedback-task/HeaderFeedbackTask.component.vue:14]: Same here

File: frontend/components/commons/header/header-feedback-task/HeaderFeedbackTask.component.vue [hunk 5]
```diff
@@ -11,7 +11,7 @@
...
+        v-if="showTrainButton"
       >
         <svgicon name="code" width="20" height="20" />Train
       </BaseButton>
```
Comment (damianpumar) [inline frontend/components/commons/header/header-feedback-task/HeaderFeedbackTask.component.vue:17]: the DatasetId on line: 24, is not used anymore

File: frontend/pages/dataset/_id/annotation-mode/index.vue [hunk 6]
```diff
@@ -7,7 +7,7 @@
           :key="datasetId"
           :datasetId="datasetId"
           :breadcrumbs="breadcrumbs"
-          :showTrainButton="true"
+          :showTrainButton="isAdminOrOwnerRole"
```
Comment (damianpumar) [inline frontend/pages/dataset/_id/annotation-mode/index.vue]: So, I think this change you removed in the next commit.

File: frontend/pages/user-settings.vue [hunk 7]
```diff
@@ -13,12 +18,22 @@
...
+import { ROUTES } from "@/v1/infrastructure/services";
 import HeaderAndTopAndTwoColumns from "@/layouts/HeaderAndTopAndTwoColumns";
 
 export default {
   name: "UserSettingsPage",
   components: {
     HeaderAndTopAndTwoColumns,
   },
+  created() {
+    this.breadcrumbs = [
```
Comment (damianpumar) [inline frontend/pages/user-settings.vue:30]: This attribute is not used...

File: frontend/components/feedback-task/header/load-line/LoadLine.vue [hunk 8]
```diff
@@ -0,0 +1,54 @@
+<!--
```
Comment (damianpumar) [inline frontend/components/feedback-task/header/load-line/LoadLine.vue:1]: what's the difference between load-line vs loading-line ?

File: frontend/translation/en.json [hunk 9]
```diff
@@ -50,8 +50,8 @@
     "recordsHeight": "Records height:",
     "fixedHeight": "Fixed height (300px)",
     "defaultHeight": "Default height",
-    "to_annotate_record_bulk_required": "To annotate you must select at least one record",
-    "records_to_be_applied": "the action will be applied to 1 record | the action will be applied to {count} records"
+    "to_annotate_record_bulk_required": "No record selected",
+    "records_to_be_applied": "1 record selected | {count} records selected"
```
Comment (damianpumar) [inline frontend/translation/en.json]: We have duplicated value on line: 48

File: frontend/components/feedback-task/header/load-line/LoadLine.vue (hunk 8 above)
Comment (leiyre) [reply frontend/components/feedback-task/header/load-line/LoadLine.vue:1]: I've "duplicated" it to open it to possible changes, do you think it's better to extend and use `LoadingLine`?

File: frontend/translation/en.json (hunk 9 above)
Comment (leiyre) [reply frontend/translation/en.json]: true! good catch

File: frontend/components/feedback-task/container/fields/RecordFieldsHeader.vue [hunk 10]
```diff
@@ -47,21 +47,11 @@ export default {
...
   methods: {
     onSelectedRecord(isSelected) {
       this.$emit("on-select-record", isSelected, this.record);
     },
   },
-  created() {
-    this.$nuxt.$on("animate-checkboxes", () => {
-      this.animateCheckboxes = true;
-    });
-  },
```
Comment (leiyre) [inline frontend/components/feedback-task/container/fields/RecordFieldsHeader.vue]: If we remove the events, the animation will be shown every time the record is rendered and we only want it to be shown when the bulk component is rendered

**Pull request merged**